oidc-service-impl/src/main/java/io/mosip/esignet/services/OAuthServiceImpl.java

```diff
@@ -104,9 +104,10 @@ public TokenResponse getTokens(TokenRequest tokenRequest,boolean isV2) throws Es
 
         authenticateClient(tokenRequest, clientDetailDto,isV2);
 
+        String userInfoResponseType=transaction.getUserInfoResponseType();
         boolean isTransactionVCScoped = isTransactionVCScoped(transaction);
         if(!isTransactionVCScoped) { //if transaction is not VC scoped, only then do KYC exchange
-            KycExchangeResult kycExchangeResult = doKycExchange(transaction);
+            KycExchangeResult kycExchangeResult = doKycExchange(transaction,userInfoResponseType);
```

Comment:
If oidcTransaction itself has the responseType, why should be send it as an different argument?